projects/ng-components/lib/tooltip/tooltip.directive.ts

```diff
@@ -1,234 +1,177 @@
-import { ComponentRef, Directive, ElementRef, HostListener, Input, OnInit, TemplateRef, HostBinding, OnDestroy } from "@angular/core";
+import { ComponentRef, Directive, ElementRef, HostListener, Input, TemplateRef, OnDestroy, Renderer2 } from "@angular/core";
 import {
     Overlay,
-    OverlayPositionBuilder,
     OverlayRef,
-    ConnectedPosition,
     PositionStrategy,
-    FlexibleConnectedPositionStrategy,
+    ConnectionPositionPair,
+    OverlayConfig,
     ConnectedOverlayPositionChange,
+    FlexibleConnectedPositionStrategy,
 } from "@angular/cdk/overlay";
 import { ComponentPortal } from "@angular/cdk/portal";
-
-import { TooltipContentComponent, TooltipTrigger, TooltipPosition, TooltipTheme } from "./tooltip-content/tooltip-content.component";
-import { Subscription } from "rxjs";
-import { distinctUntilChanged } from "rxjs/internal/operators/distinctUntilChanged";
+import { TooltipContentComponent, TooltipTrigger, TooltipTheme } from "./tooltip-content/tooltip-content.component";
+import { Subject, fromEvent } from "rxjs";
+import { distinctUntilChanged, takeUntil } from "rxjs/operators";
 import { isEqual } from "lodash";
```

Comment:
We don't use lodash as a dependency, and it may not exists in a project using our component. Maybe you can just write the function inside the component itself, it is much safer that way

Reply:
removed it, using Object.keys instead since we do have control over the object type and order of keys:
```javascript
(prev: ConnectedOverlayPositionChange, curr: ConnectedOverlayPositionChange) => Object.keys(prev) === Object.keys(curr)
```